posts/2021-06-17-Rust-1.53.0.md

```diff
@@ -0,0 +1,180 @@
+including the long-awaited `IntoIterator` implementation for arrays.
+See the [detailed release notes](https://github.com/rust-lang/rust/blob/master/RELEASES.md#version-1530-2021-06-17)
+to learn about other changes not covered by this post.
+
+### IntoIterator for arrays
+
+This is the first Rust release in which arrays implement the `IntoIterator` trait.
+This means you can now iterate over arrays by value:
+
+```rust
+for i in [1, 2, 3] {
+    ..
+}
+```
+
+Previously, this was only possible by reference, using `&[1, 2, 3]` or `[1, 2, 3].iter()`.
+
+Similarly, you can now pass arrays to methods expecting a `T: IntoIterator`:
+
+```rust
+let set = BTreeSet::from_iter([1, 2, 3]);
+```
+
+```rust
+for (a, b) in some_iterator.chain([1]).zip([1, 2, 3]) {
+    ..
+}
+```
+
+This was not implemented before, due to backwards compatibility problems.
+Because `IntoIterator` was already implemented for references to arrays,
+`array.into_iter()` already compiled in earlier versions,
+resolving to `(&array).into_iter()`.
+
+As of this release, arrays implement `IntoIterator` with a small workaround to avoid breaking code.
+The compiler will continue to resolve `array.into_iter()` to `(&array).into_iter()`,
+as if the trait implementation does not exist.
+This only applies to the `.into_iter()` method call syntax, and does not
+affect any other syntax such as `for e in [1, 2, 3]`, `iter.zip([1, 2, 3])` or
+`IntoIterator::into_iter([1, 2, 3])`, which all compile fine.
+
+Since this special case for `.into_iter()` is only required to avoid breaking existing code,
+it is removed in the new edition, Rust 2021, which will be released later this year.
+See [the edition announcement](https://blog.rust-lang.org/2021/05/11/edition-2021.html#intoiterator-for-arrays)
+for more information.
+
+### Or patterns
+
+Pattern syntax has been extended to support `|` nested anywhere in the pattern.
+This enables you to write `Some(1 | 2)` instead of `Some(1) | Some(2)`.
+
+```rust
+match result {
+     Ok(Some(1 | 2)) => { .. }
+     Err(MyError { kind: FileNotFound | PermissionDenied, .. }) => { .. }
+     _ => { .. }
+}
+```
+
+### Unicode identifiers
+
+Identifiers can now contain non-ascii characters.
+All valid identifier characters in Unicode as defined in [UAX #31](https://unicode.org/reports/tr31/) can now be used.
+
+For example:
+
+```rust
+const BLÅHAJ: &str = "🦈";
+
+struct 人 {
+    名字: String,
+}
+
+let α = 1;
+```
+
+The compiler will warn about about potentially confusing situations involving different scripts.
+For example, using identifiers that look very similar will result in a warning.
+
+```
+warning: identifier pair considered confusable between `ｓ` and `s`
+```
+
+### HEAD branch name support in Cargo
+
+Cargo no longer assumes the default `HEAD` of git repositories is named `master`.
+This means you no longer need to specify `branch = "main"` for git dependencies
+from a repository where the main branch is called `main`.
```

Comment:
It may be nice to match what github calls it:
```suggestion
from a repository where the default branch is called `main`.
```
![image](https://user-images.githubusercontent.com/17426603/121012418-fdfae700-c797-11eb-93cd-a37e72c54818.png)